src/test/scala/is/hail/methods/LinearRegression3Suite.scala

```diff
@@ -0,0 +1,194 @@
+    val v9 = Variant("1", 9, "C", "T") // x = (., 1, 1, 1, 1, 1)
+    val v10 = Variant("1", 10, "C", "T") // x = (., 2, 2, 2, 2, 2)
+
+    val qBeta = vds.queryVA("va.linreg.beta")._2
+    val qSe = vds.queryVA("va.linreg.se")._2
+    val qTstat = vds.queryVA("va.linreg.tstat")._2
+    val qPval = vds.queryVA("va.linreg.pval")._2
+
+    val annotationMap = vds.variantsAndAnnotations
+      .collect()
+      .toMap
+
+    def assertInt(q: Querier, v: Variant, value: Int) =
+      assert(D_==(q(annotationMap(v)).asInstanceOf[Int], value))
+
+    def assertDouble(q: Querier, v: Variant, value: Double) = {
+      val x = q(annotationMap(v)).asInstanceOf[IndexedSeq[Double]].apply(0)
+      assert(D_==(x, value))
+    }
+
+    /*
+    comparing to output of R code:
+    y = c(1, 1, 2, 2, 2, 2)
+    x = c(0.009900990296049406, 0.9900990100009803, 0.009900990296049406, 0.009900990296049406, 0.009900990296049406, 0.9900990100009803)
+    c1 = c(0, 2, 1, -2, -2, 4)
+    c2 = c(-1, 3, 5, 0, -4, 3)
+    df = data.frame(y, x, c1, c2)
+    fit <- lm(y ~ x + c1 + c2, data=df)
+    summary(fit)["coefficients"]
+
+    */
+
+    assertDouble(qBeta, v1, -0.29166985)
+    assertDouble(qSe, v1, 1.2996510)
+    assertDouble(qTstat, v1, -0.22442167)
+    assertDouble(qPval, v1, 0.84327106)
+
+    /*
+    v2 has two missing genotypes, comparing to output of R code as above with imputed genotypes:
+    x = c(0.9950495050004902, 1.980198019704931, 0.9950495050004902, 1.980198019704931, 0.009900990296049406, 0.009900990296049406)
+    */
+
+    assertDouble(qBeta, v2, -0.5499320)
+    assertDouble(qSe, v2, 0.3401110)
+    assertDouble(qTstat, v2, -1.616919)
+    assertDouble(qPval, v2, 0.24728705)
+
+    /*
+    v3 has two missing genotypes, comparing to output of R code as above with imputed genotypes:
+    x = c(0.009900990296049406, 0.7450495050747477, 0.9900990100009803, 0.9900990100009803, 0.9900990100009803, 0.7450495050747477)
+    */
+
+    assertDouble(qBeta, v3, 1.09536219)
+    assertDouble(qSe, v3, 0.6901002)
+    assertDouble(qTstat, v3, 1.5872510)
+    assertDouble(qPval, v3, 0.2533675)
+  }
+
+  @Test def testBgen12FastPath() {
+    hc.indexBgen("src/test/resources/example.8bits.bgen")
+    val vds = hc.importBgen("src/test/resources/example.8bits.bgen")
+      // linreg3_sa has random normal pheno1, pheno2, cov1, cov2 with a small amount of missingness
+      .annotateSamplesTable(hc.importTable("src/test/resources/linreg3_sa.tsv", keyNames = Array("Sample"), impute = true), root = "sa")
+      .linreg3(Array("sa.pheno1", "sa.pheno2"), Array("sa.cov1", "sa.cov2"), useDosages = true, root = "va.linreg3")
+      .linregMultiPheno(Array("sa.pheno1", "sa.pheno2"), Array("sa.cov1", "sa.cov2"), useDosages = true, root = "va.multi")
+      // lingreg3 is defined and agrees with multi when it is defined
+      .annotateVariantsExpr(
+        "va.close = isMissing(va.multi) || (va.linreg3.beta - va.multi.beta).forall(d => d < 0.001)")
```

Comment:
I think you want `(isMissing(va.multi) && isMissing(va.linreg3)` since we should worry if linreg3 is defined but multi is not